tests/e2e/tests/pilot/performance/serviceentry_test.go

```diff
@@ -0,0 +1,328 @@
+package performance
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/gogo/protobuf/types"
+	"github.com/google/uuid"
+	"github.com/onsi/gomega"
+
+	mcp "istio.io/api/mcp/v1alpha1"
+	networking "istio.io/api/networking/v1alpha3"
+	mixerEnv "istio.io/istio/mixer/test/client/env"
+	"istio.io/istio/pilot/pkg/bootstrap"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/adsc"
+	"istio.io/istio/pkg/mcp/snapshot"
+	mcptest "istio.io/istio/pkg/mcp/testing"
+	"istio.io/istio/tests/util"
+)
+
+var (
+	serviceEntryCount = flag.Int("serviceEntryCount", 1000, "Number of Service Entries to register")
+	terminateAfter    = flag.Int("terminateAfter", 60, "Number of seconds to run the test")
```

Comment:
Use `flag.Duration` here so that `terminateAfter` is of type `time.Duration`. This will simplify line 142 which you could move inline to line 93.